packages/rum-core/src/common/url.js

```diff
@@ -136,6 +136,20 @@ class Url {
 
     this.protocol = protocol || location.protocol
 
+    /**
+     * Construct port and hostname from host and assign defaults based
+     * on http and https protocols
+     */
+    if (/:\d+$/.test(this.host)) {
+      const value = this.host.split(':')
```

Comment:
perhaps use capturing groups in the regex? that might also simplify removing square brackets from IPv6 addresses

Reply:
Decided to not use capture groups and went with explicit ipv6 check to keep things simpler.